Automattic/jetpack
# [Boost] Switch critical CSS datastores to React

Converts all Critical CSS and related data sources over to React hooks and DataSync / actions.

Notes:
- I'm still unhappy with how the `CloudCssMeta` and `CriticalCssMeta` components share a lot of code inefficiently. I'm not happy with the props passed to `<Status>` and how they're used. But this PR is large enough, so I will address those in a future PR.
- I've noticed taht speed scores currently refresh while the generator is running. That seems to be a bug on the speed-score debounce side of things we'll want to fix in another PR.

## Proposed changes:
* Convert all Critical CSS data sources to React hooks.

### Other information:

## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


no

## Testing instructions:





* Test out both local and cloud critical css generation.
* Make some pages fail and ensure errors are shown properly.
